server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -33,40 +40,62 @@
  * A field mapper that records fields that have been ignored because they were malformed.
  */
 public final class IgnoredFieldMapper extends MetadataFieldMapper {
-
     public static final String NAME = "_ignored";
-
     public static final String CONTENT_TYPE = "_ignored";
 
     public static class Defaults {
         public static final String NAME = IgnoredFieldMapper.NAME;
 
         public static final FieldType FIELD_TYPE = new FieldType();
+        public static final FieldType LEGACY_FIELD_TYPE = new FieldType();
 
         static {
             FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
             FIELD_TYPE.setTokenized(false);
             FIELD_TYPE.setStored(true);
             FIELD_TYPE.setOmitNorms(true);
             FIELD_TYPE.freeze();
+
+            LEGACY_FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
+            LEGACY_FIELD_TYPE.setTokenized(false);
+            LEGACY_FIELD_TYPE.setStored(false);
+            LEGACY_FIELD_TYPE.setOmitNorms(true);
+            LEGACY_FIELD_TYPE.freeze();
         }
     }
 
-    public static final TypeParser PARSER = new FixedTypeParser(c -> new IgnoredFieldMapper());
+    public static final TypeParser PARSER = new FixedTypeParser(c -> new IgnoredFieldMapper(c.indexVersionCreated()));
 
     public static final class IgnoredFieldType extends StringFieldType {
 
+        public static final IgnoredFieldType INSTANCE_LEGACY = new IgnoredFieldType(false);
         public static final IgnoredFieldType INSTANCE = new IgnoredFieldType();
 
         private IgnoredFieldType() {
-            super(NAME, true, false, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+            this(true);
+        }
+
+        private IgnoredFieldType(boolean docValues) {
+            super(NAME, true, docValues, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+        }
+
+        public static IgnoredFieldType getInstance(Version version) {
+            return version.onOrAfter(Version.V_8_0_0)
```

Comment:
Given that we refer to this version in a few different places, I think it's worth adding a static value directly on the Mapper that we can use?
```
public static void Version DOC_VALUES_VERSION = Version.V_8_0_0
```

Reply:
I agree